### ISSUE TITLE
Why is Dice metric computed with data rather than pred in _val_step?

### ISSUE THREAD

**Issue by Hmxki** [OSUPCVLab/SegFormer3D#30]
In the _val_step function of SegFormer3D/train_scripts/trainer_ddp.py, I noticed that the Dice score is computed using
mean_dice = self._calc_dice_metric(data, labels, use_ema) (line 213).

Could you please clarify why data is used as the input for Dice calculation here? My understanding is that Dice is usually computed based on the model predictions (e.g., pred) rather than the input data itself.

I might be missing something in the code flow, so any explanation would be greatly appreciated. Thank you!

**Comment by bnavard** [comment]
If you take a look at the `SlidingWindowInference ` implementation [here](https://github.com/OSUPCVLab/SegFormer3D/blob/main/metrics/segmentation_metrics.py#L66) (line 66), you’ll see that we pass a predictor argument to `sliding_window_inference`. This predictor is simply the model itself. Internally, MONAI’s `sliding_window_inference` calls the model via its `__call__ ` method to run inference on each window.

Because of that design, this metric expects the raw input tensor rather than precomputed predictions—the predictions are generated internally during the sliding-window pass. That’s why the API looks a bit different from metrics that operate directly on logits or probability maps.

That said, you’re absolutely right that this could have been designed more cleanly from a software-engineering perspective 😄. We appreciate you pointing it out, and we’ll consider improving the interface to make this behavior more explicit.

Hope this clears things up, and thanks again for the feedback!
